### PR TITLE
Add changes due to native ballerina based streaming impl improvements

### PR DESCRIPTION
**Pull request by mohanvive**

### CODE REVIEW

File: website/two-column-pages/docs/learn/ballerina-streaming-reference.md [hunk 1]
```diff
@@ -785,6 +786,143 @@ Following are the supported operations of a join clause.
     ```
 
 
+#### Table Operations
+Ballerina provides extensive support to deal with tables. It provides various types of operations such as create, insert, delete and etc.. with in-memory or external storage tables. You could refer below link to find more information on this.
```
Comment (samgnaniah) [inline website/two-column-pages/docs/learn/ballerina-streaming-reference.md]:
```suggestion
Ballerina provides extensive support to deal with tables. It provides various types of operations such as create, insert, delete, etc. with in-memory or external storage tables.
```

I don't see the link so I'm deleting that line for now.

File: website/two-column-pages/docs/learn/ballerina-streaming-reference.md [hunk 2]
```diff
@@ -785,6 +786,143 @@ Following are the supported operations of a join clause.
...
+}
+
+```
+
+
+****
+
+Capabilities mentioned in below sections are not supported by the native Ballerina based
+stream processing. They are only available when using Siddhi CEP engine for stream processing capabilities
+in Ballerina. If you wanted to enable Siddhi runtime based stream processing with Ballerina; then please use
```
Comment (samgnaniah) [inline website/two-column-pages/docs/learn/ballerina-streaming-reference.md]:
```suggestion
in Ballerina. If you want to enable Siddhi runtime-based stream processing with Ballerina; please use
```

File: website/two-column-pages/docs/learn/ballerina-streaming-reference.md [hunk 3]
```diff
@@ -785,6 +786,143 @@ Following are the supported operations of a join clause.
...
+    }
+    return result;
+}
+
+```
+
+
+****
+
+Capabilities mentioned in below sections are not supported by the native Ballerina based
```
Comment (samgnaniah) [inline website/two-column-pages/docs/learn/ballerina-streaming-reference.md]: Not sure which "below sections" we are referring to here.

File: website/two-column-pages/docs/learn/ballerina-streaming-reference.md [hunk 4]
```diff
@@ -785,6 +786,143 @@ Following are the supported operations of a join clause.
...
+        select item.name as itemName, item.stockAmount
+        => (OutOfStockAlert[] alerts) {
+            foreach var alert in alerts {
+                oredrAlertStream.publish(alert);
+            }
+        }
+    }
+}
+
+//`queryItemTable` function returns a table of items whose stock is not enough to satisfy the order.
```
Comment (samgnaniah) [inline website/two-column-pages/docs/learn/ballerina-streaming-reference.md]: Back ticks do not render in code blocks in ballerina.io. Better remove these.

File: website/two-column-pages/docs/learn/ballerina-streaming-reference.md [hunk 5]
```diff
@@ -785,6 +786,143 @@ Following are the supported operations of a join clause.
...
+```ballerina
+//This is the record that holds item details in the stockTable.
+type Item record {
+    string name;
+    float price;
+    int stockAmount;
+    !...
+};
+
+//This is the record that holds order events from customer.
```
Comment (samgnaniah) [inline website/two-column-pages/docs/learn/ballerina-streaming-reference.md]:
```suggestion
// This is the record that holds order events from the customer.
```

Can we add spaces before all comments in this doc?

File: website/two-column-pages/docs/learn/ballerina-streaming-reference.md [hunk 6]
```diff
@@ -785,6 +786,143 @@ Following are the supported operations of a join clause.
...
 
+#### Table Operations
+Ballerina provides extensive support to deal with tables. It provides various types of operations such as create, insert, delete and etc.. with in-memory or external storage tables. You could refer below link to find more information on this.
+
+###### Purpose
+In Streaming context, a table is a stored version of a stream or a table of events. Ballerina provides support to interactively query the state of the stored events in the table when processing events which are arrived through a stream.
+We could perform operations such as add, delete, update and join with tables.
+
+###### Example - Add
+In below query events which arrive in `stockStream` are added into the table `itemStockTable` after projecting few attributes from the event.
```
Comment (samgnaniah) [inline website/two-column-pages/docs/learn/ballerina-streaming-reference.md]:
```suggestion
In the following example, query events that arrive in `stockStream` are added into the table `itemStockTable` after projecting a few attributes from the event.
```

File: website/two-column-pages/docs/learn/ballerina-streaming-reference.md [hunk 7]
```diff
@@ -785,6 +786,143 @@ Following are the supported operations of a join clause.
...
+            }
+        }
+    }
+}
+
+```
+
+
+###### Example - Join with Table
+In the following query, we are performing a join operation between the event stream and table. Whenever an order event is published to `orderStream`, it is matched against the `itemStockTable` through the `queryItemTable` function. If there is a match,
```
Comment (samgnaniah) [inline website/two-column-pages/docs/learn/ballerina-streaming-reference.md]:
```suggestion
In the following query, we perform a join operation between the event stream and table. Whenever an order event is published to `orderStream`, it is matched against the `itemStockTable` through the `queryItemTable` function. If there is a match,
```

File: website/two-column-pages/docs/learn/ballerina-streaming-reference.md (hunk 4 above)
Comment (mohanvive) [reply website/two-column-pages/docs/learn/ballerina-streaming-reference.md]: Actually, it works right. It gets highlighted. Isn't it?

**Review by Manuri — APPROVED** [review]

**Pull request merged**